src/ComposerRepository.php

```diff
@@ -53,6 +64,12 @@ protected function fetchFile($filename, $cacheKey = null, $sha256 = null, $store
             return [];
         }
 
-        return parent::fetchFile($filename, $cacheKey, $sha256, $storeLastModifiedTime);
+        $data = parent::fetchFile($filename, $cacheKey, $sha256, $storeLastModifiedTime);
+
+        if (0 === strpos($filename, 'http://packagist.org/p/symfony/') || 0 === strpos($filename, 'https://packagist.org/p/symfony/')) {
```

Comment:
Would it make more sense if we used regular expressions like `^https?:\/\/packagist\.com\/p\/symfony`?

Reply:
From PHP documentation : `Do not use preg_match() if you only want to check if one string is contained in another string. Use strpos() instead as it will be faster.`

Reply:
It'd make more sense to either use preg_match or parse_url and check for the domain being packagist.org OR repo.packagist.org as we are likely going to switch to that domain in the future.

Reply:
Actually, do we care about the domain? We should discard the legacy tags even if a proxy or a private Packagist is used.